hax/test/integration/test_server.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from base64 import b64encode
+from typing import List
+
+import pytest
+import simplejson
+from hax.log import setup_logging
+from hax.message import BroadcastHAStates
+from hax.motr import WorkPlanner
+from hax.motr.delivery import DeliveryHerald
+from hax.server import ServerRunner
+from hax.types import Fid, HAState, MessageId, ServiceHealth
+from hax.util import ConsulUtil
+
+
+@pytest.fixture(autouse=True)
+def logging_support():
+    setup_logging()
+    yield ''
+    logging.shutdown()
+
+
+@pytest.fixture
+def herald(mocker):
+    return DeliveryHerald()
+
+
+@pytest.fixture
+def planner(mocker) -> WorkPlanner:
+    def fake_add(cmd):
+        if hasattr(cmd, 'reply_to') and cmd.reply_to:
+            cmd.reply_to.put([MessageId(1, 42)])
+
+    planner = WorkPlanner()
+    mocker.patch.object(planner, 'add_command', side_effect=fake_add)
+    return planner
+
+
+@pytest.fixture
+def consul_util(mocker):
+    consul = ConsulUtil()
+    exc = RuntimeError('Not allowed')
+    mock = mocker.patch.object
+    mock(consul.kv, 'kv_get', side_effect=exc)
+    mock(consul.kv, 'kv_put', side_effect=exc)
+    mock(consul, 'get_local_nodename', side_effect=['localhost'])
+    mock(consul, 'get_hax_ip_address', side_effect=['192.168.0.28'])
+    return consul
+
+
+@pytest.fixture
+async def hax_client(mocker, aiohttp_client, herald, planner, consul_util,
+                     loop):
+    srv = ServerRunner(planner, herald, consul_util)
+    srv._configure()
+    return await aiohttp_client(srv.app)
+
+
+async def test_hello_works(hax_client):
+    resp = await hax_client.get('/')
+    assert resp.status == 200
+    text = await resp.text()
+    assert text == "I'm alive! Sincerely, HaX"
+
+
+@pytest.mark.parametrize('status,health', [('passing', ServiceHealth.OK),
```

Comment:
This seems like a decorator and would pass all these statuses one by one to the decorated function is it? If so, its pretty cool.

Reply:
Yes, it is the way to feed test function with different inputs and provide expected outputs. And thus save dozens of lines of code :-)